datastream-server/src/main/java/com/linkedin/datastream/server/DummyTransportProviderAdminFactory.java

```diff
@@ -28,6 +28,7 @@
   public static final String PROVIDER_NAME = "default";
 
   private final boolean _throwOnSend;
+  private boolean _failTransportProvider;
```

Comment:
Looks like you want to fail the transport provider only once. You can change the name to `_failTransportProviderOnce`.